app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java

```diff
@@ -2152,6 +2152,38 @@ public void forgotPassword_validEmailNotRegistered_emailNotSent() {
         assertTextOnScreen(EMAIL_DOES_NOT_EXIST);
     }
 
+    @Test
+    public void autoCompleteTextView_PartOfValidUsername_usernameSelected() {
+        logInTestUser();
+
+        // Asserts that the current activity is the MainActivity. Otherwise, show “Wrong Activity”
+        _solo.assertCurrentActivity(WRONG_ACTIVITY_MESSAGE, MainActivity.class);
+
+        // click on the Notifications tab in the bottom navigation bar
+        _solo.clickOnView(_solo.getView(R.id.navigation_notifications));
+
+        // ensure that the app has transitioned to the Profile screen
+        assertTextOnScreen(PROFILE_TEXT);
+
+        // click follow user search button
+        _solo.clickOnView(_solo.getView(R.id.follow_user_search_button));
+
+        // click on the auto complete text view
+        _solo.clickOnView(_solo.getView(R.id.auto_complete_text_view));
+
+        // enter part of the users username we wish to follow
+        _solo.enterText(_solo.getEditText("Username"), TEST_USER_SUBSTRING);
+
+        // click on the user name that we wish to follow
+        // Click on text does not work for item in autoCompleteTextView
+        // so I decided to go with click on screen at a specific location
+        _solo.clickOnScreen(540, 1170);
```

Comment:
Setting hardcoded X and Y values in a UI test isn't a good approach. Sure this may pass on your emulator, but it could fail on other devices. I see the comment you've left, but this is just asking for a flaky test. I found this [link](https://stackoverflow.com/questions/19860782/how-to-test-for-autocompletetextview-itemclick-event-using-robotium). You can click on the text, you just have to wait for it to appear first.

- [ ] remove hardcoded numbers from UI test (if possible)

Reply:
I tried this approach as well and I found it wouldn't pass the test so I ended up not using it